### DISCUSSION TITLE
Scaffold existing DB

### DISCUSSION THREAD

**Discussion by juniorschroder** [adonisjs/core#1254] [Ideas]
Will be Nice and helpfull to port existing projects to adonisjs, a command in ace to connect database and generate models

**Comment by thetutlage** [comment]
I thought about it, not only the generating models, even updating them with new columns. So ideally, you always write migrations by hand and then run `node ace update:models` to get them in sync with your database.

But there are couple of issues with this

- Need to store `table -> model` mappings somewhere (I usually hate generating these sort of manifest files)
- Retaining the file formatting as it was earlier. So we began by generating the AST, update the AST and then write it back as a string. However, the existing tools like `tsc` or [recast](https://www.npmjs.com/package/recast) are not 100% perfect in retaining the file formatting.

**Comment by thetutlage** [comment]
If anyone, who has interests in working with AST can ping me here and I can get them up and running by sharing the idea and small prototype I have around this.

**Reply by juniorschroder** [reply]
I can try help with this

**Reply by thetutlage** [reply]
Cool. Give me a day or so and then I will share all the research and maybe some concrete steps on how to proceed forward :)

**Reply by shemsiu** [reply]
I can try to help as well :)

**Reply by unolee** [reply]
I can try to help.
